reader.go

```diff
@@ -0,0 +1,52 @@
+package kinesumer
+
+import (
+	"io"
+
+	k "github.com/remind101/kinesumer/interface"
+)
+
+// Reader provides an io.Reader implementation that can read data from a kinesis
+// stream.
+type Reader struct {
+	// buf is used internally to buffer up received messages.
+	buf []byte
+
+	records <-chan k.Record
+}
+
+// NewReader returns a new Reader instance that reads data from records.
+func NewReader(records <-chan k.Record) *Reader {
+	return &Reader{records: records}
+}
+
+// Read implements io.Reader Read.
+func (r *Reader) Read(p []byte) (n int, err error) {
```

Comment:
Is it expected that consumers call Done() on every record or is it only necessary to call it on the last record?